media/js/base/global.js

```diff
@@ -34,15 +34,22 @@ function init_download_links() {
 }
 
 function update_download_text_for_old_fx() {
-    // if using an out of date firefox
-    if (isFirefox() && !isFirefoxUpToDate()) {
-        // look at each button to see if it's set to check for old firefox
-        $('.download-button').each(function() {
-            var $button = $(this);
-
-            if ($button.hasClass('download-button-check-old-fx')) {
-                // replace subtitle copy
-                $button.find('.download-subtitle').text(window.trans('global-update-firefox'));
+    var $buttons = $('.download-button');
+
+    // if using Firefox and the page has download buttons
+    if (isFirefox() && $buttons.length) {
```

Comment:
It may not be expensive in the grand scheme of things, but it is wasteful taking into account the high number of pages that don't need this user information.